src/reducers/clauseTemplatesReducer.js

```diff
@@ -42,6 +44,20 @@ const reducer = (state = initialState, action) => {
         ...state,
         [action.clauseTemplate.id]: action.clauseTemplate,
       };
+    case REMOVE_CLAUSE_TEMPLATE:
+      let filteredTemplateClauses;
+      for (const [key, value] of Object.entries(state)) {
+        if (value.clauseId === action.clauseId) {
+          filteredTemplateClauses = R.omit(
+            [key],
+            state
+          );
+        }
+      }
+      return {
+        ...filteredTemplateClauses
+      };
```

Comment:
I think this would be a more efficient approach:

```js
return R.omit(
        (R.slice(0, 1, Object.entries(state).find(
          ([key, value]) => (value.clauseId === action.clauseId)
        ) || []) || []),
        state
      );
```